### ISSUE TITLE
RFC: new rule to prefer the usage of screen

### ISSUE THREAD

**Issue by timdeschryver** [testing-library/eslint-plugin-testing-library#95]
It's preferred to use `screen` when using the query functions.

```js
// bad
const component = render()
component.getBy()

// good
screen.getBy()
```

For a first implementation we can verify that the queries used have `screen` as callee.
This doesn't take the following into account, but it makes the rule simpler.

```
const screen = render()
screen.getBy
```

**Comment by Belco90** [comment]
I guess we can skip that last case for now. What about?

```
const { getByText } = render();
const button = getByText('submit');
```

I think this one should be included too, which would be queries without a calle?

**Comment by timdeschryver** [comment]
Imho these would also be a warning.

**Comment by Belco90** [comment]
Cool, so we can create a new rule called `prefer-screen-queries` which report all queries not called from `screen.<query>`.

**Comment by timdeschryver** [comment]
👍 I'll try to implement this rule later this week

**Comment by Belco90** [comment]
That would be awesome :)

**Comment by timdeschryver** [comment]
Closed by #99
